packages/purpose-process/test/testGetPurposes.ts

```diff
@@ -0,0 +1,450 @@
+  getMockPurpose,
+  writeInReadmodel,
+  getMockValidRiskAnalysisForm,
+} from "pagopa-interop-commons-test/index.js";
+import { genericLogger } from "pagopa-interop-commons";
+import { addOnePurpose, getMockEService } from "./utils.js";
+import {
+  postgresDB,
+  purposes,
+  eservices,
+  purposeService,
+} from "./purposeService.integration.test.js";
+
+export const testGetPurposes = (): ReturnType<typeof describe> =>
+  describe("getPurposes", async () => {
+    const producerId1: TenantId = generateId();
+    const producerId2: TenantId = generateId();
+    const consumerId1: TenantId = generateId();
+
+    const mockEService1ByTenant1: EService = {
+      ...getMockEService(),
+      producerId: producerId1,
+    };
+
+    const mockEService2ByTenant1: EService = {
+      ...getMockEService(),
+      producerId: producerId1,
+    };
+
+    const mockEService3ByTenant2: EService = {
+      ...getMockEService(),
+      producerId: producerId2,
+    };
+
+    const mockEService4 = getMockEService();
+
+    const mockPurposeVersion1: PurposeVersion = {
+      ...getMockPurposeVersion(),
+      state: purposeVersionState.draft,
+    };
+    const mockPurpose1: Purpose = {
+      ...getMockPurpose(),
+      title: "purpose 1 - test",
+      consumerId: consumerId1,
+      eserviceId: mockEService1ByTenant1.id,
+      versions: [mockPurposeVersion1],
+    };
+
+    const mockPurposeVersion2: PurposeVersion = {
+      ...getMockPurposeVersion(),
+      state: purposeVersionState.draft,
+    };
+    const mockPurpose2: Purpose = {
+      ...getMockPurpose(),
+      title: "purpose 2",
+      eserviceId: mockEService1ByTenant1.id,
+      versions: [mockPurposeVersion2],
+    };
+
+    const mockPurposeVersion3: PurposeVersion = {
+      ...getMockPurposeVersion(),
+      state: purposeVersionState.active,
+    };
+    const mockPurpose3: Purpose = {
+      ...getMockPurpose(),
+      title: "purpose 3",
+      eserviceId: mockEService2ByTenant1.id,
+      versions: [mockPurposeVersion3],
+    };
+
+    const mockPurposeVersion4: PurposeVersion = {
+      ...getMockPurposeVersion(),
+      state: purposeVersionState.rejected,
+    };
+    const mockPurpose4: Purpose = {
+      ...getMockPurpose(),
+      title: "purpose 4",
+      eserviceId: mockEService3ByTenant2.id,
+      versions: [mockPurposeVersion4],
+    };
+
+    const mockPurposeVersion5: PurposeVersion = {
+      ...getMockPurposeVersion(),
+      state: purposeVersionState.rejected,
+    };
+    const mockPurpose5: Purpose = {
+      ...getMockPurpose(),
+      title: "purpose 5",
+      consumerId: consumerId1,
+      eserviceId: mockEService4.id,
+      versions: [mockPurposeVersion5],
+    };
+
+    const mockPurposeVersion6_1: PurposeVersion = {
+      ...getMockPurposeVersion(),
+      state: purposeVersionState.archived,
+    };
+    const mockPurposeVersion6_2: PurposeVersion = {
+      ...getMockPurposeVersion(),
+      state: purposeVersionState.draft,
```

Comment:
This is currently not possible, a purpose version can be `draft` only when it's the first version, you can set to any other state without any problems.

Reply:
I needed it to evaluate the `excludeDraft` filter.

Reply:
> I needed it to evaluate the `excludeDraft` filter.

It works already if it can exclude `mockPurpose1` and `mockPurpose2`, IMHO there's no need to test a case that is not compliant with our data model

Reply:
But the implementation of the check is split into two parts. Purpose with no versions or with only one draft version are filtered inside the query. After that, purposes with both draft and non draft versions get the draft versions removed after the query. (Reference: [first](https://github.com/pagopa/interop-be-monorepo/pull/438/files#diff-3151b523d08fc3302f8957cb874cae400f744fa16160fab31d47d7a5c2854f3cR175-R190) part, [second](https://github.com/pagopa/interop-be-monorepo/pull/438/files#diff-95b4e078c342343517e86a98ea5650ede6c4ec6e5c9f4547dbfb44c8d6d22118R545-R549) part).
Accordingly to your point, should I removed the second part entirely? cc @galales

Reply:
> But the implementation of the check is split into two parts. Purposes with no versions or with only one draft version are filtered inside the query. After that, purposes with both draft and non-draft versions get the draft versions removed after the query. (Reference: [first](https://github.com/pagopa/interop-be-monorepo/pull/438/files#diff-3151b523d08fc3302f8957cb874cae400f744fa16160fab31d47d7a5c2854f3cR175-R190) part, [second](https://github.com/pagopa/interop-be-monorepo/pull/438/files#diff-95b4e078c342343517e86a98ea5650ede6c4ec6e5c9f4547dbfb44c8d6d22118R545-R549) part). According to your point, should I remove the second part entirely? cc @galales

Yes I think that the second part isn't needed if we assume that a version in `Draft` exists only if the only one present

Reply:
I agree that it is not required if it is a condition that should never occur.
@taglioni-r could you please link the reference in Scala coda where this is done? I may be missing some previous considerations

Reply:
Sorry, I might have gone too straightforward in following what we did on `getEservices`. Indeed in this case there's not such logic. Removed.